### ISSUE TITLE
Batch delete and batch update without batching?

### ISSUE THREAD

**Issue by RudeySH** [zzzprojects/EntityFramework-Plus#642]
Is it possible to use `.Delete()` or `.Update(x => ...)` without the batching? I would like to be able to delete or update entities without loading them into memory, but I don't want EFPlus to generate a query that will perform a batch delete/update with a loop.

**Comment by JonathanMagnan** [comment]
Hello @RudeySH ,

Are you using EF6 or EF Core? We will look at it.

Best Regards,

Jon

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by JonathanMagnan** [comment]
We quickly looked at the code, if you use EF Core that's already possible by setting the `BatchSize = 0`.

Could you look at it and let me know if that work for you?

**Comment by RudeySH** [comment]
I'm using EF6. I couldn't find any documentation on `BatchSize = 0`, so I'm not sure if this also works for EF6.

**Comment by JonathanMagnan** [comment]
Hello @RudeySH ,

A new version is available (v1.12.37),

We added the support to `BatchSize = 0` for EF6 to avoid creating a batch size

```csharp
context.EntitySimples.Delete(option => option.BatchSize = 0);
```

Let me know if that was what you were looking for.

Best Regards,

Jon
